src/sqlfluff/api/simple.py

```diff
@@ -38,7 +77,13 @@ def lint(
     Returns:
         :obj:`list` of :obj:`dict` for each violation found.
     """
-    linter = Linter(dialect=dialect, rules=rules, exclude_rules=exclude_rules)
+    cfg = get_simple_config(
+        dialect=dialect,
+        rules=rules,
+        exclude_rules=exclude_rules,
+        config_path=config_path,
+    )
+    linter = Linter(config=cfg)
```

Comment:
we can hopefully remove those dialect/rules/exclude_rules arguments to the linter in the future as they should be part of the config. i.e. decouple linter and simple API 👍

Reply:
That would mean you'd have to have a config file to pass. Might be overkill for some simple options. For example I presume https://online.sqlfluff.com/ uses the simple API and just needs to specify the dialect.

But it is duplication which is annoying.

BTW what happens if you specify a dialect AND a config file? Which ones wins out? Or should that error and it should be either/or but not both?

Reply:
@tunetheweb you can still pass a dialect to the Simple API (and the dialect would overwrite that in the config as you'd expect)? 
The Simple API itself is unchanged from a user perspective:
```python
res = sqlfluff.lint(
    dialect="snowflake",
    rules=["L001", "L002"],
    config_path="some/path/to/config/.sqlfluff"
)
```

I'm specifically talking about `Linter` in the comment which doesn't need those arguments 😄

Reply:
as in we don't need to be able to do `Linter(dialect="snowflake")` since the simple api was the only thing using that. Everything else does `Linter(config=cfg)`.

Reply:
Ah Gothca.

Yes In agree explicit dialect or rule should override config, so that's good, but I wonder if we need to expose this in the docs though so others know that's the way it's implemented?